numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -1891,8 +1891,16 @@ array_scalar(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *kwds)
                 &PyArrayDescr_Type, &typecode, &obj)) {
         return NULL;
     }
+    if (PyDataType_FLAGCHK(typecode, NPY_LIST_PICKLE)) {
+        if (!PySequence_Check(obj)) {
+            PyErr_SetString(PyExc_TypeError,
+                            "scalar pickle not returning sequence");
```

Comment:
Ah, now your test is there I understand the purpose of this error. Perhaps:
```suggestion
                            "found non-sequence while unpicking scalar with "
                            "NPY_LIST_PICKLE set");
```